src/main/java/bio/terra/app/configuration/CacheConfiguration.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.app.configuration;
+
+import com.google.common.cache.CacheBuilder;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CachingConfigurer;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.concurrent.ConcurrentMapCache;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+import org.springframework.cache.interceptor.CacheErrorHandler;
+import org.springframework.cache.interceptor.CacheResolver;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.cache.interceptor.SimpleCacheErrorHandler;
+import org.springframework.cache.interceptor.SimpleCacheResolver;
+import org.springframework.cache.interceptor.SimpleKeyGenerator;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Configures a cache to be used by Spring.  It likely makes sense to use a cache per endpoint.
+ */
+@EnableCaching
+@Configuration
+public class CacheConfiguration implements CachingConfigurer {
+
+    public static final String SPRING_CACHE_MANAGER = "springCacheManager";
+
+    // Use these strings to reference CacheProperties
+    public static final String STATUS_PROP = "STATUS";
+
+    public enum CacheProperties {
+        STATUS(30L, TimeUnit.SECONDS, null);
+
+        private final long duration;
+        private final TimeUnit unit;
+        private final Optional<Long> maxSize;
+
+        CacheProperties(Long duration, TimeUnit unit, Long maxSize) {
```

Comment:
Since `duration` can't be `null` I would use an unboxed type here
```suggestion
        CacheProperties(long duration, TimeUnit unit, Long maxSize) {
```

Reply:
Fair enough. I was hoping to keep it consistent but that makes sense